training/src/app/store/generateJoke.actions.ts

```diff
@@ -0,0 +1,9 @@
+import { createAction, props } from '@ngrx/store';
+import { Joke } from "../chuck-noris/joke";
+
+export const generateJoke        = createAction('generateJoke');
+export const generateJokeSuccess = createAction('generateJoke success', props<{ payload: Joke }>());
+export const generateJokeFailed  = createAction('generateJoke failed', props<{ payload: string }>());
+export const getJokeList         = createAction('getJokeList', props<{ payload: any }>());
```

Comment:
why the payload isnt a string?

Reply:
![image](https://user-images.githubusercontent.com/92257009/158343626-bf76c344-d29a-4070-bad3-d5b5f8263675.png)
here is the error i get when trying to change to string